yogstation/code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -0,0 +1,272 @@
+	nojumpsuit = 1
+	mutanteyes = /obj/item/organ/eyes/night_vision/alien/sling
+	burnmod = 1.5 //1.5x burn damage, 2x is excessive
+	heatmod = 1.5
+	var/mutable_appearance/eyes_overlay
+	var/shadow_charges = 3
+	var/last_charge = 0
+
+/datum/species/shadow/ling/on_species_gain(mob/living/carbon/human/C)
+	C.draw_yogs_parts()
+	eyes_overlay = mutable_appearance('yogstation/icons/mob/sling.dmi', "eyes", 25)
+	C.add_overlay(eyes_overlay)
+	. = ..()
+
+/datum/species/shadow/ling/on_species_loss(mob/living/carbon/human/C)
+	C.draw_yogs_parts(TRUE)
+	if(eyes_overlay)
+		C.cut_overlay(eyes_overlay)
+		QDEL_NULL(eyes_overlay)
+	. = ..()
+
+/datum/species/shadow/ling/spec_life(mob/living/carbon/human/H)
+	var/light_amount = 0
+	H.nutrition = NUTRITION_LEVEL_WELL_FED //i aint never get hongry
+	if(isturf(H.loc))
+		var/turf/T = H.loc
+		light_amount = T.get_rgb_lumcount(r_mul = LIGHT_RED_MULTIPLIER, g_mul = LIGHT_GREEN_MULTIPLIER, b_mul = LIGHT_BLUE_MULTIPLIER)
+		if(light_amount > LIGHT_DAM_THRESHOLD) //Can survive in very small light levels. Also doesn't take damage while incorporeal, for shadow walk purposes
+			H.take_overall_damage(0, LIGHT_DAMAGE_TAKEN)
+			if(H.stat != DEAD)
+				to_chat(H, "<span class='userdanger'>The light burns you!</span>") //Message spam to say "GET THE FUCK OUT"
+				H.playsound_local(get_turf(H), 'sound/weapons/sear.ogg', 150, 1, pressure_affected = FALSE)
+		else if (light_amount < LIGHT_HEAL_THRESHOLD)
+			H.heal_overall_damage(5,5)
+			H.adjustToxLoss(-5)
+			H.adjustBrainLoss(-25) //Shad O. Ling gibbers, "CAN U BE MY THRALL?!!"
+			H.adjustCloneLoss(-1)
+			H.SetKnockdown(0)
+			H.SetStun(0)
+	var/charge_time = 400 - ((SSticker.mode.thralls && SSticker.mode.thralls.len) || 0)*10
+	if(world.time >= charge_time+last_charge)
+		shadow_charges = min(shadow_charges + 1, 3)
+		last_charge = world.time
+
+/datum/species/shadow/ling/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	var/turf/T = H.loc
+	if(istype(T) && shadow_charges > 0)
+		var/light_amount = T.get_rgb_lumcount(r_mul = LIGHT_RED_MULTIPLIER, g_mul = LIGHT_GREEN_MULTIPLIER, b_mul = LIGHT_BLUE_MULTIPLIER)
+		if(light_amount < LIGHT_DAM_THRESHOLD)
+			H.visible_message("<span class='danger'>The shadows around [H] ripple as they absorb \the [P]!</span>")
+			playsound(T, "bullet_miss", 75, 1)
+			shadow_charges = min(shadow_charges - 1, 0)
+			return -1
+	return 0
+
+/datum/species/shadow/ling/lesser //Empowered thralls. Obvious, but powerful
+	name = "Lesser Shadowling"
+	id = "l_shadowling"
+	say_mod = "chitters"
+	species_traits = list(NOBLOOD,NO_DNA_COPY,NOTRANSSTING,NOEYES)
+	inherent_traits = list(TRAIT_NOBREATH, TRAIT_RADIMMUNE)
+	burnmod = 1.1
+	heatmod = 1.1
+
+/datum/species/shadow/ling/lesser/spec_life(mob/living/carbon/human/H)
+	var/light_amount = 0
+	H.nutrition = NUTRITION_LEVEL_WELL_FED //i aint never get hongry
+	if(isturf(H.loc))
+		var/turf/T = H.loc
+		light_amount = T.get_lumcount()
```

Comment:
Delete the ``var/light_amount = 0`` at the start of the proc and make this ``var/light_amount = ...``.